documentation/metadata-editing.md

```diff
@@ -0,0 +1,119 @@
+
+### Protected metadata
+
+Some metadata is forbidden to be modified.
+The CSS will throw an error (409 `ConflictHttpError`) when trying to change this protected metadata.
+
+## Impact on creating containers
+
+Before editing description resources was possible, it was possible to change the description resource of `ldp:Container`s.
+This was achieved by executing a `PUT` request to the container and adding the metadata triples in the body of that request.
+
+This was not in line with the [Solid Protocol](https://solidproject.org/TR/2021/protocol-20211217#writing-resources)
+about writing resources.
+The protocol states the following:
+
+> Servers MUST NOT allow HTTP POST, PUT and PATCH to update a container’s resource metadata statements;
+> if the server receives such a request, it MUST respond with a 409 status code.
+
+Therefore, since it is now possible to edit description resources directly,
+the above functionality has been removed from the CSS.
+This means that now when creating a container, the contents of the body are ignored.
+Additionally, it is now forbidden to perform a `PUT` request to an existing container.
+
+## Example of a workflow for editing a description resource {#workflow}
+
+In this example, we are trying to add an inbox description to an `ldp:Resource`.
+This inbox indication will then allow discovery for the `ldp:inbox` as described in the [Linked Data Notifications specification](https://www.w3.org/TR/ldn/).
+
+To give some context, assume we have started the CSS with the default configuration
+and have created an inbox which has as URL `http://localhost:3000/inbox/`.
+
+We don't know the location of the description resource
+So first, we send a `HEAD` request to the resource to retrieve the description resource URL.
+
+```shell
+curl --location --head 'http://localhost:3000/'
+```
+
+which will at least the following response
+
+```shell
+HTTP/1.1 200 OK
+Link: <http://localhost:3000/.meta>; rel="describedby"
+```
+Now that we have the URL of the description resource,
+we create a query for adding the inbox in the description of the resource.
+This can be either with a SPARQL Update
+
+```shell
+curl --location --request PATCH 'http://localhost:3000/.meta' \
+--header 'Content-Type: application/sparql-update' \
+--data-raw 'INSERT DATA { <http://localhost:3000/> <http://www.w3.org/ns/ldp#inbox> <http://localhost:3000/inbox/>. }'
+```
+
+or with an N3 Patch.
+
+```shell
+curl --location --request PATCH 'http://localhost:3000/resource.meta' \
+--header 'Content-Type: text/n3' \
+--data-raw '@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+<> a solid:InsertDeletePatch;
+solid:inserts { <http://localhost:3000/> <http://www.w3.org/ns/ldp#inbox> <http://localhost:3000/inbox/>. }.'
+```
+
+After this update, we can verify that the inbox is added by performing a GET request to the description resource
+
+```shell
+curl --location --request GET 'http://localhost:3000/.meta'
+```
+With as result for the body
```

Comment:
It should also be in the response header right?

Reply:
Not when performing the GET to the description resource.
When you do a GET request to the subject resource, it is in the response header (due the configuration in the metadata-writer).

Reply:
Ah true, perhaps mention that as well then.